rfc/2019-12-10-add-pulsar-to-binlog.md

```diff
@@ -0,0 +1,39 @@
+# Proposal: Add Apache Pulsar to tidb-binlog
+
+- Author(s): Xiaolong Ran
+- Last updated:  Dec 05, 2019
+- Discussion at: https://github.com/pingcap/community/issues/117
+
+## Abstract
+
+The proposal aims at:
+
+Use **key_shared** subscription type of [Apache Pulsar](https://github.com/apache/pulsar) to enhance **binlog** data processing capabilities;
+
+## Background
+
+During the Change Data Capture(CDC), we need to ensure the order of the messages, or, we need to ensure that DDL arrives before DML. Currently, we use Kafka to process the order of messages, but as you know, Kafka can only guarantee the order of messages within a single partition. If we need to expand the downstream data processing capabilities, sometimes we want more partitions to provide more processing power, but how do we ensure the order of messages at this time? This is a trickier question.
+
+## Proposal
+
+We will make these changes:
+
+1. Replace **kafak-client-go** with [pulsar-client-go](https://github.com/apache/pulsar-client-go).
+2. Provide a plugin to support more message queues, including: Apache Pulsar
```

Comment:
a plugin?  What does it mean? Is there a more detailed explanation?

Reply:
We can make an abstraction of the existing interface implementation, aka, expose an interface, Kafka and Pulsar can implement this interface. 

In this way, all Message Queues that implement this interface can be integrated into tidb-binlog, right?

Reply:
The rfc's most direct purpose is to replace Kafka with Pulsar. 

In here, I was wondering if we have a more general way to take into account the compatibility of the version and make less intrusion into the original code.

Reply:
The rfc's most direct purpose is to replace Kafka with Pulsar. 

In here, I was wondering if we have a more general way to take into account the compatibility of the version and make less intrusion into the original code.

Reply:
right, same with [my comment above](https://github.com/pingcap/community/pull/136#discussion_r358647539) , just do it